packages/block-library/src/query-title/edit.js

```diff
@@ -47,6 +48,33 @@ export default function QueryTitleEdit( {
 		titleElement = (
 			<TagName { ...blockProps }>{ __( 'Archive title' ) }</TagName>
 		);
+	} else if ( type === 'search' ) {
+		titleElement = (
+			<>
+				<InspectorControls>
+					<PanelBody title={ __( 'Settings' ) }>
+						<ToggleControl
+							label={ __( 'Show search term in title' ) }
+							onChange={ () =>
+								setAttributes( {
+									showSearchTerm: ! showSearchTerm,
+								} )
+							}
+							checked={ showSearchTerm }
+						/>
+					</PanelBody>
+				</InspectorControls>
+				{ showSearchTerm ? (
```

Comment:
We probably don't need to repeat `<TagName..` here, we could replace this with a smaller statement:

```
<TagName { ...blockProps }>
	{ showSearchTerm
		? __( 'Search results for: "search term"' )
		: __( 'Search results:' ) }
</TagName>
```